crypto/err/err.c

```diff
@@ -228,8 +228,8 @@ static void build_SYS_str_reasons(void)
 
                 str->string = cur;
                 cnt += l;
-                if (cnt > sizeof(strerror_pool))
-                    cnt = sizeof(strerror_pool);
+                if (cnt >= sizeof(strerror_pool))
```

Comment:
This should not be necessary, or is the wrong place to fix the buffer overrun.
If `cnt >= sizeof(strerror_pool)` then the string did already exceed the array bounds.

Reply:
The ```cnt >``` part shouldn't happen but the ```cnt == ``` does, when then end of strerror_pool is reached.  
However I noticed that even with my commit the ```cur``` pointer happily grows and writes zeros after the end of strerror_pool. I'll fix it and update the pull request.